### PR TITLE
Added methods to encode and decode messages to and from dicts

### PR DESCRIPTION
**Pull request by hackaugusto**
This can be used to json encode the messages

### CODE REVIEW

File: raiden/messages.py [hunk 1]
```diff
@@ -995,6 +1116,40 @@ def from_event(event):
...
+            'lock': self.lock.to_dict(),
+            'locksroot': self.locksroot.hex(),
+            'target': self.target.hex(),
+            'initiator': self.initiator.hex(),
+            'fee': self.fee,
+            'signature': self.signature.hex(),
+        }
+
+    @staticmethod
+    def form_dict(data):
```
Comment (czepluch) [inline raiden/messages.py]: also here

File: raiden/messages.py [hunk 2]
```diff
@@ -939,6 +1026,40 @@ def from_event(event: 'SendMediatedTransfer') -> 'MediatedTransfer':
...
```
Comment (czepluch) [inline raiden/messages.py]: I don't like the form of this "from"

Comment (LefterisJP) [reply raiden/messages.py]: sshh don't spoil it :)

File: raiden/messages.py (hunk 1 above)
Comment (hackaugusto) [reply raiden/messages.py]: I left some behind to check you were paying attention

**Comment by LefterisJP** [comment]
@hackaugusto Can you also maybe put a test for the `from/to` dict functions?

**Comment by hackaugusto** [comment]
will do

**Comment by hackaugusto** [comment]
@LefterisJP added some tests, see if you think these are sufficient. The tests are not checking the encoded value, only that after encode/decode cycle that the values are properly recovered.

**Pull request merged**